### PR TITLE
Set up JS Standards

### LINKED ISSUE #26: JS Standards
### Description

Establish standards for working in the JS application, patterns that promote best practicies.

### User Story

As a developer I want to follow best practices  in JS to ensure the application is maintainable by additional team members.

### Architectural Decision Records (ADRs)

_No response_

### PR DESCRIPTION
**Pull request by scottqueen-bixal**
This establishes JS Standards for the react application in our BEARS v2 project.

The following updates include;

1. README.md file
2. Directory structure
3. Directory descriptions
4. Githooks (husky) to interact with automated standards
5. Linters and formaters, eslint, js-standard, react, stylelint, prettier
6. Node scripts to interact with standards
7. Guidance for best practice on code working agreements
8. Guidance on typechecking
9. Guidance on styles
10. Guidance on code documentation
11. Guidance on testing, jest
12. Guidance on using a generating new components

Related Github Issue
- Fixes #26 

Testing

@FatmaBakir
- [ ] review readme file

@nehemiah-abuga
- [x] review readme file

@gchi25 since this is code heavy, please take the following steps to pressure test this work

- [ ] review readme file, ensure that all aspects of document make sense from a developer perspective
- [ ] clone branch to your local
- [ ] cd into bears-app
- [ ] npm install`
- [ ] ensure that husky githooks are installed
- [ ] test pre-commit githook
- [ ] test pre-push githook
- [ ] review linters that are included and their custom configuration/overides within the application
- [ ] test linters and formatting scripts
- [ ] review working agreement for code pattern avoidance
- [ ] review proptypes documentation and in code
- [ ] review sass and sass modules guidance
- [ ] review JSDoc and inline documentation of code
- [ ] review Jest, run tests
- [ ] review plop, test generating a new component

### CODE REVIEW

File: bears-app/README.md [hunk 1]
```diff
@@ -1,70 +1,245 @@
...
 
 This project was bootstrapped with [Create React App](https://github.com/facebook/create-react-app).
 
-## Available Scripts
+Get our packages.
 
-In the project directory, you can run:
+```shell
+npm run install
+```
```
Comment (gchi25) [inline bears-app/README.md:51]: npm install

File: bears-app/README.md [hunk 2]
```diff
@@ -1,70 +1,245 @@
...
+npm run format
+```
 
-### Making a Progressive Web App
+Run stylelint on `.scss` files
 
-This section has moved here: [https://facebook.github.io/create-react-app/docs/making-a-progressive-web-app](https://facebook.github.io/create-react-app/docs/making-a-progressive-web-app)
+```shell
+npm run lint:scss
+```
```
Comment (gchi25) [inline bears-app/README.md:109]: npm run lint:scss 
> bears-app@0.1.0 lint:scss 
> stylelint 'src/**/*.scss' --syntax scss 
Invalid option "--syntax". Did you mean "--config"?

File: bears-app/README.md [hunk 3]
```diff
@@ -1,70 +1,245 @@
...
 
-### Advanced Configuration
+Auto-fix lint errors
 
-This section has moved here: [https://facebook.github.io/create-react-app/docs/advanced-configuration](https://facebook.github.io/create-react-app/docs/advanced-configuration)
+```shell
+npm run lint:scss:fix
+```
```
Comment (gchi25) [inline bears-app/README.md:115]: npm run lint:scss:fix 
> bears-app@0.1.0 lint:scss:fix 
> stylelint 'src/**/*.scss' --syntax scss --fix 
Invalid option "--syntax". Did you mean "--config"?

**Review by gchi25 — COMMENTED** [review]
Reviewed readme file with comment.
Did clone branch to your local; cd into bears-app; npm install`
Tested npm start. http://localhost:3000 worked as expected.
husky githooks are installed.
Tested pre-commit githook. Worked as expected.
Tested  test pre-push githook. Worked as expected.
Tested linters and formatting scripts with issues as commented.
Reviewed working agreement for code pattern avoidance
Reviewed proptypes documentation and in code
Reviewed sass and sass modules guidance
Reviewed JSDoc and inline documentation of code
Tested Jest. Worked as expected.
Tested generating new component. Worked as expected.

File: bears-app/README.md (hunk 1 above)
Comment (scottqueen-bixal) [reply bears-app/README.md:51]: updated

File: bears-app/README.md (hunk 2 above)
Comment (scottqueen-bixal) [reply bears-app/README.md:109]: updated

File: bears-app/README.md (hunk 3 above)
Comment (scottqueen-bixal) [reply bears-app/README.md:115]: updated

**Review by gchi25 — APPROVED** [review]
Reviewed readme file.
Tested linters. Worked as expected.

**Review by FatmaBakir — APPROVED** [review]
I reviewed the README files and all look good to me.

**Review by nehemiah-abuga — APPROVED** [review]
Reviewed the readme. Looking good

**Pull request merged**